File: capibara-website/app/models/Function.scala
package models
import play.api.libs.json._

object Return {
  implicit val returnFormatter = Json.format[Return];
}

object Parameter {
  implicit val parameterFormatter = Json.format[Parameter];
}

object Example {
  implicit val exampleFormatter = Json.format[Example];
}

object Function {
  implicit val functionFormatter = Json.format[Function];
}

case class Function(
    name: String,
    header: HeaderRef,
    summary: String,
    returns: Return,
    parameters: Seq[Parameter],
    description: String,
    examples: Seq[Example],
    associated: Seq[String],
    os_affinity: Seq[String]
)

case class Return(
    `type`: String,
    description: String,
)

case class Parameter(
    name: String,
    `type`: String,
    description: String,
)

case class Example(
    title: String,
    code: String,
)

File: capibara-website/src/models/function.rs
use super::header::HeaderSummary;
use rocket::serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Clone)]
pub(crate) struct Function {
    pub name: Box<String>,
    pub header: HeaderSummary,
    pub summary: Box<String>,
    pub returns: Return,
    pub parameters: Vec<Parameter>,
    pub description: Box<String>,
    pub examples: Vec<Example>,
    pub associated: Vec<String>,
    pub os_affinity: Vec<String>,
}

#[derive(Serialize, Deserialize, Clone)]
pub(crate) struct Example {
    pub title: Box<String>,
    pub code: Box<String>,
}

#[derive(Serialize, Deserialize, Clone)]
pub(crate) struct Parameter {
    pub name: Box<String>,
    #[serde(rename = "type")]
    pub _type: Box<String>,
    pub description: Box<String>,
}

#[derive(Serialize, Deserialize, Clone)]
pub(crate) struct Return {
    #[serde(rename = "type")]
    pub _type: Box<String>,
    pub description: Box<String>,
}
